test/Microsoft.AspNetCore.WebSockets.Server.Test/Autobahn/AutobahnTester.cs

```diff
@@ -0,0 +1,144 @@
+                if (result.ExitCode != 0)
+                {
+                    throw new Exception("wstest failed, stderr: " + result.StdErr);
+                }
+            }
+            finally
+            {
+                if (File.Exists(specFile))
+                {
+                    File.Delete(specFile);
+                }
+            }
+
+            // Parse the output.
+            var outputFile = Path.Combine(Directory.GetCurrentDirectory(), Spec.OutputDirectory, "index.json");
+            using (var reader = new StreamReader(File.OpenRead(outputFile)))
+            {
+                return AutobahnResult.FromReportJson(JObject.Parse(await reader.ReadToEndAsync()));
+            }
+        }
+
+        public void Verify(AutobahnResult result)
+        {
+            var failures = new StringBuilder();
+            foreach(var serverResult in result.Servers)
+            {
+                var serverExpectation = _expectations.FirstOrDefault(e => e.Server == serverResult.Server && e.Ssl == serverResult.Ssl);
+                if(serverExpectation == null)
+                {
+                    failures.AppendLine($"Expected no results for server: {serverResult.Name} but found results!");
+                }
+                else
+                {
+                    serverExpectation.Verify(serverResult, failures);
+                }
+            }
+
+            Assert.True(failures.Length == 0, "Autobahn results did not meet expectations:" + Environment.NewLine + failures.ToString());
+        }
+
+        public async Task DeployTestAndAddToSpec(ServerType server, bool ssl, string environment, Action<AutobahnExpectations> expectationConfig = null)
+        {
+            var port = Interlocked.Increment(ref _nextPort);
+            var baseUrl = ssl ? $"https://localhost:{port}" : $"http://localhost:{port}";
+            var sslNamePart = ssl ? "SSL" : "NoSSL";
+            var name = $"{server}|{sslNamePart}|{environment}";
+            var logger = _loggerFactory.CreateLogger($"AutobahnTestApp:{server}:{sslNamePart}:{environment}");
+
+            var appPath = Helpers.GetApplicationPath("AutobahnTestApp");
+            var parameters = new DeploymentParameters(appPath, server, RuntimeFlavor.CoreClr, RuntimeArchitecture.x64)
+            {
+                ApplicationBaseUriHint = baseUrl,
+                ApplicationType = ApplicationType.Portable,
+                TargetFramework = "netcoreapp1.0",
+                EnvironmentName = environment
+            };
+
+            var deployer = ApplicationDeployerFactory.Create(parameters, logger);
+            var result = deployer.Deploy();
+
+#if NET451
+            System.Net.ServicePointManager.ServerCertificateValidationCallback = (_, __, ___, ____) => true;
+            var client = new HttpClient();
+#else
+            var handler = new HttpClientHandler()
+            {
+                ServerCertificateCustomValidationCallback = (_, __, ___, ____) => true
```

Comment:
Ok. The simplest thing for me to do here would be to disable SSL tests on Mac. AFAIK Linux would work though since libcurl there does use OpenSSL.

We're not trying to test all SSL scenarios here, just scenarios we'd expect to see in production. And in our case, if Kestrel's SSL stuff is behaving (i.e. your tests @mikeharder ;P), our stuff should be working, this is just a sanity check.